### ISSUE TITLE
enforce-canonical: variant-prefixed opacity forms (dark:bg-white/[.08]) not autofixed

### ISSUE THREAD

**Issue by Nayeem-XTREME** [sergioazoc/oxlint-tailwindcss#156]
## Summary

`tailwindcss/enforce-canonical` correctly rewrites bare utilities like `bg-white/[.08]` → `bg-white/8`, but **does not report or autofix** the same utility when a variant is present (`dark:bg-white/[.08]`, `hover:bg-white/[.08]`, `sm:dark:bg-white/[.08]`).

Tailwind CSS IntelliSense (`suggestCanonicalClasses`) still flags these.

## Environment

- `oxlint-tailwindcss@1.12.0`
- `oxlint@1.83.0`
- `tailwindcss@4.3.3`

## Reproduction

```jsx
export default function Example() {
  return (
    <>
      <div className="bg-white/[.08]" />
      <div className="bg-black dark:bg-white/[.08]" />
      <div className="bg-white hover:bg-black/[.06]" />
    </>
  );
}
```

With `enforce-canonical: "error"` and a valid `settings.tailwindcss.entryPoint`:

| Class | Reported |
|-------|----------|
| `bg-white/[.08]` | yes → `bg-white/8` |
| `dark:bg-white/[.08]` | **no** |
| `hover:bg-black/[.06]` | **no** |

`oxlint --fix` leaves the variant-prefixed forms untouched.

## Root cause (observed)

`canonicalizeCandidates('dark:bg-white/[.08]')` **does** return `dark:bg-white/8`.

The worker `safe` gate compares `candidatesToCss` via `declsOf()`, which uses `indexOf('{')` / `lastIndexOf('}')`. For variant output nested in `@media` / `:hover`, that slice includes the **selector** (whose escaped class name differs), so `safe` becomes `false` and the rewrite is dropped.

For the bare utility, decls-only equality holds → autofix applies.

Utility-only canonicalize + utility-only decls compare is safe:

```js
declsOf('bg-white/[.08]') === declsOf('bg-white/8') // true
```

Then reattach the variant prefix: `dark:` + `bg-white/8`.

## Suggested fix

In `enforce-canonical`, for classes with a variant prefix and a dynamic utility:

1. `splitUtilityAndVariant`
2. `canonicalizeCandidates([utility])` (and safe-check utility decls only)
3. Rebuild `variant + canonicalUtility` (preserve `!`)

Or harden `declsOf` to extract the innermost declaration block so nested `@media` selectors are ignored.

## Workaround

We shipped a small local oxlint JS plugin that does utility-first canonicalize + reattach until this lands upstream.

**Comment by sergioazoc** [comment]
Thanks for the detailed report and the root-cause analysis — it was spot on. The fix is in #163.

It turned out to be broader than the opacity forms: no arbitrary-value rewrite ever applied under a variant that wraps the rule in an at-rule (`hover:z-[10]` → `hover:z-10` was dropped too). Instead of canonicalizing the utility and reattaching the variant, the safe check now neutralizes only the class's own name in the emitted CSS and compares everything else, so the value-preserving guarantee from #78 still holds.

While auditing, we also found that arbitrary variants were never canonicalized (`data-[open]:` → `data-open:`, `min-[40rem]:` → `sm:`); that's fixed in the same PR. It will ship in the next release, and your local workaround plugin can be dropped then.
